packages/odyssey-react-mui/src/labs/TopNav.tsx

```diff
@@ -0,0 +1,459 @@
+})(({ odysseyDesignTokens }: { odysseyDesignTokens: DesignTokens }) => ({
+  color: odysseyDesignTokens.TypographyColorHeading,
+  fontSize: odysseyDesignTokens.TypographySizeSubordinate,
+}));
+
+const UserProfileOrgInfoContainer = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})(({ odysseyDesignTokens }: { odysseyDesignTokens: DesignTokens }) => ({
+  color: odysseyDesignTokens.TypographyColorSubordinate,
+  fontSize: odysseyDesignTokens.TypographySizeSubordinate,
+}));
+
+const UserProfile = ({ profileIcon, userName, orgName }: UserProfileProps) => {
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+
+  return (
+    <UserProfileContainer odysseyDesignTokens={odysseyDesignTokens}>
+      {profileIcon && (
+        <UserProfileIcon odysseyDesignTokens={odysseyDesignTokens}>
+          {profileIcon}
+        </UserProfileIcon>
+      )}
+      <UserProfileInfoContainer>
+        <UserProfileEmailInfoContainer
+          odysseyDesignTokens={odysseyDesignTokens}
+        >
+          {userName}
+        </UserProfileEmailInfoContainer>
+        <UserProfileOrgInfoContainer odysseyDesignTokens={odysseyDesignTokens}>
+          {orgName}
+        </UserProfileOrgInfoContainer>
+      </UserProfileInfoContainer>
+    </UserProfileContainer>
+  );
+};
+
+const TopNavListContainer = styled.ul({
+  padding: 0,
+  listStyle: "none",
+  listStyleType: "none",
+  display: "flex",
+  alignItems: "center",
+});
+
+const TopNavItemLabelContainer = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})<{
+  odysseyDesignTokens: DesignTokens;
+}>(({ odysseyDesignTokens }) => ({
+  display: "flex",
+  flexWrap: "wrap",
+  alignItems: "center",
+  fontSize: odysseyDesignTokens.TypographyScale0,
+  fontWeight: odysseyDesignTokens.TypographyWeightHeading,
+}));
+
+const TopNavListItemContainer = styled("li", {
+  shouldForwardProp: (prop) =>
+    prop !== "odysseyDesignTokens" && prop !== "isDisabled",
+})<{
+  odysseyDesignTokens: DesignTokens;
+  isDisabled?: boolean;
+}>(({ odysseyDesignTokens, isDisabled }) => ({
+  display: "flex",
+  alignItems: "center",
+  cursor: isDisabled ? "default" : "pointer",
+  pointerEvents: isDisabled ? "none" : "auto",
+  "& a": {
+    display: "flex",
+    alignItems: "center",
+    padding: `${odysseyDesignTokens.Spacing2} ${odysseyDesignTokens.Spacing4}`,
+    color: `${odysseyDesignTokens.TypographyColorHeading} !important`,
+  },
+  "& a:hover": {
+    textDecoration: "none",
+    backgroundColor: !isDisabled ? odysseyDesignTokens.HueNeutral50 : "inherit",
+  },
+  "& div[role='button']:hover": {
+    backgroundColor: !isDisabled ? odysseyDesignTokens.HueNeutral50 : "inherit",
+  },
+  "& a:focus-visible": {
+    outlineOffset: 0,
+    borderRadius: 0,
+    outlineWidth: odysseyDesignTokens.FocusOutlineWidthMain,
+    backgroundColor: !isDisabled ? odysseyDesignTokens.HueNeutral50 : "inherit",
+  },
+}));
```

Comment:
Since we're always doing `!isDisabled`, would it make sense to use `isEnabled` instead? It'd make the code easier to read, but I understand it doesn't follow the way HTML does it.

Something to think about. I prefer readability in these cases since these are regular `div`s and not form elements.